eth_utils/conversions.py

```diff
@@ -0,0 +1,157 @@
+# String encodings and numeric representations
+import json
+
+from cytoolz import (
+    curry,
+)
+
+from .decorators import (
+    assert_one_arg,
+)
+from .encoding import (
+    big_endian_to_int,
+    int_to_big_endian,
+)
+from .hexidecimal import (
+    add_0x_prefix,
+    decode_hex,
+    encode_hex,
+    is_hex,
+    remove_0x_prefix,
+)
+from .types import (
+    is_boolean,
+    is_dict,
+    is_integer,
+    is_string,
+)
+
+
+@assert_one_arg
+def to_hex(value=None, hexstr=None, text=None):
+    """
+    Auto converts any supported value into it's hex representation.
+
+    Trims leading zeros, as defined in:
+    https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding
+    """
+    if hexstr is not None:
+        return add_0x_prefix(hexstr.lower())
+
+    if text is not None:
+        return encode_hex(text.encode('utf-8'))
+
+    if is_boolean(value):
+        return "0x1" if value else "0x0"
+
+    if is_dict(value):
+        return encode_hex(json.dumps(value, sort_keys=True))
```

Comment:
This is undocumented functionality.

And also odd functionality that I'm curious where we are using it and if we can just remove it...

Reply:
`git blame` says it's very old (last change from a year ago), so I'll just remove it and double-check by running the full test suite in web3.py. https://github.com/ethereum/web3.py/pull/584